elephant/conversion.py

```diff
@@ -152,3 +155,679 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+            spike_idx.append(l)
+        return spike_idx
+
+    def to_bool_array(self):
+        """
+        Returns a dense matrix (`scipy.sparse.csr_matrix`), which rows
+        represent the number of spike trains and the columns represent the
+        binned index position of a spike in a spike train.
+        The matrix columns contain only ones, which indicate a spike.
+
+        Returns
+        -------
+        bool matrix : numpy.ndarray
+            Returns a dense matrix representation of the sparse matrix,
+            with ones indicating **True** for a spike and **False*** for
+            non spike.
+            The **Trues** in the columns represent the index
+            position of the spike in the spike train and rows represent the
+            number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> import quantities as pq
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.TimeHistogram(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.to_bool_array())
+        [[ True  True False  True  True  True  True False False False]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+        """
+        return abs(scipy.sign(self.to_array())).astype(bool)
+
+    def to_array(self):
+        """
+        Returns a dense matrix, calculated from the sparse matrix with counted
+        time points, which rows represents the number of spike trains and the
+        columns represents the binned index position of a spike in a
+        spike train.
+        The  matrix columns contain the number of spikes that
+        occurred in the spike train(s).
+        If **bool** :attr:`store_mat` is set to **True** last calculated
+        matrix will be returned.
+
+        Returns
+        -------
+        matrix : numpy.ndarray
+            Matrix with spike times. Columns represent the index position of
+            the binned spike and rows represent the number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.TimeHistogram(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.to_array())
+        [[2 1 0 1 1 1 1 0 0 0]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+
+        """
+        if self._mat_u is not None:
+            return self._mat_u
+        if self._store_mat_u:
+            self._mat_u = self.to_sparse_array().toarray()
+            return self._mat_u
+        # Matrix on demand
+        else:
+            return self._sparse_mat_u.toarray()
+
+    def store_array(self):
```

Comment:
> store_array = False seems pretty explicit to me, but ok. I like your proposal, but shouldn't the argument be added to to_array and to_bool_array, not to the "sparse" variants? The class always stores the sparse array anyway, the "store_array" argument is for the dense array, no

Yes, sorry I meant the `to_array`. 
Besides that, I also had implemented a method for storing the boolean version of the sparse matrix. I will remove that.  

Let me summarize: The class calculates internally the **sparse non boolean** matrix when instantiated. 
There exists one method to : 
* get the non boolean matrix - `to_sparse_array`
* calculate the **boolean** matrix out of the non boolean and return it - `to_sparse_bool_array`
* return a dense matrix non boolean version - `to_array` (will get a parameter to store)
* store non boolean version in memory - `store_array` (will be removed)
* remove non boolean version from memory - `remove_stored_array`
* store boolean version in memory - `store_sparse_bool_array` (will be removed)
* remove boolean version - `remove_stored_sparse_bool_array` (will be removed)   
What do you think about the last two points (will be removed)? Should I remove them, so no storing of boolean sparse matrix?